### PR TITLE
Update to 0.9.0

### PR DESCRIPTION
**Pull request by Luthaf**
In addition to an update to 0.9, this PR contains a few changes of the API (660089f5d84923cd7155bd6aabc39b6c180537ed) I would like to discuss.

Previously, in order to change the name of an atom in a frame, one had to write the following:

```python
topology = frame.topology()
new_topology = Topology()

for atom in topology:
    atom.set_name("Fe")
    new_topology.add_atom(atom)

frame.set_topology(topology)
```

Where every function call is making a copy. The introduction of the shared allocator on the C++ side, and the use of `@property` on Python side allow to write the same code like this:

```python
for atom in frame:
    atom.name = "Fe"
```

In the same spirit, accessing all the bonds in the system is now `frame.topology.bonds` instead of `frame.topology().bonds()`.

Do you think these changes are worthwile? I think this makes the API more Python-like, at the cost of making it a bit more different from the other languages.

### CODE REVIEW

File: tests/trajectory.py [hunk 1]
```diff
@@ -99,14 +90,9 @@ def test_close(self):
 
     def test_write(self):
         frame = Frame()
-        frame.resize(4)
```
Comment (frodofine) [inline tests/trajectory.py]: Why is resize now a bad idea?

File: tests/trajectory.py [hunk 2]
```diff
@@ -16,81 +18,70 @@ def get_data_path(data):
...
-            positions[124].all(),
-            np.array([5.099554, -0.045104, 14.153846]).all()
+            frame.positions[124].all(), np.array([5.099554, -0.045104, 14.153846]).all()
         )
 
-        topology = frame.topology()
-        self.assertEqual(topology.natoms(), 297)
-        self.assertEqual(topology.atom(0).name(), "O")
-        self.assertEqual(topology.atom(1).name(), "H")
+        self.assertEqual(len(frame.atoms), 297)
```
Comment (frodofine) [inline tests/trajectory.py]: More Pythonic!

File: tests/topology.py [hunk 3]
```diff
@@ -91,38 +123,43 @@ def test_out_of_bounds(self):
         topology.resize(4)
 
         residue = Residue("")
-        residue.add_atom(1)
-        topology.add_residue(residue)
+        residue.atoms.append(1)
+        topology.residues.append(residue)
+
+        _ = topology.atoms[2]
```
Comment (frodofine) [inline tests/topology.py]: I don't understand the meaning of the `_` here. Mind explaining it?

File: chemfiles/atom.py [hunk 4]
```diff
@@ -19,121 +19,153 @@ class Atom(CxxPointer):
...
-        '''
+        """
+        if not isinstance(name, string_type):
+            raise ChemfilesError(
+                "Invalid type {} for an atomic property name".format(type(name))
+            )
         ptr = self.ffi.chfl_atom_get_property(self, name.encode("utf8"))
         return Property.from_ptr(ptr).get()
+
+    def __setitem__(self, name, value):
```
Comment (frodofine) [inline chemfiles/atom.py]: Other special functions like `__repl__` may also be useful here. I typically use Python to prototype something before rewriting the prototype into a different language and this kind of support can be useful.

File: tests/trajectory.py (hunk 1 above)
Comment (Luthaf) [reply tests/trajectory.py]: It's not that it is a bad idea, more that we can now directly add atoms to the frame, without creating a separated topology first.

File: tests/topology.py (hunk 3 above)
Comment (Luthaf) [reply tests/topology.py]: `_ = whatever()` is a way to ignore the returned value, while still making it clear that the function is returning something.

File: chemfiles/atom.py (hunk 4 above)
Comment (Luthaf) [reply chemfiles/atom.py]: Are you talking about `__repr__` ? If so yes, we could add them. I don't know how to format/represent an atom though.

File: tests/topology.py (hunk 3 above)
Comment (frodofine) [reply tests/topology.py]: Thanks!

File: chemfiles/atom.py (hunk 4 above)
Comment (Luthaf) [reply chemfiles/atom.py]: I'll do that in a separate PR/Issue

**Pull request merged**